src/Npgsql/NpgsqlConnector.cs

```diff
@@ -920,29 +920,12 @@ async Task ConnectAsync(NpgsqlTimeout timeout, CancellationToken cancellationTok
 
             Task<IPAddress[]> GetHostAddressesAsync(NpgsqlTimeout timeout, CancellationToken cancellationToken)
             {
-                // .NET 6.0 added cancellation support to GetHostAddressesAsync, which allows us to implement real
-                // cancellation and timeout. On older TFMs, we fake-cancel the operation, i.e. stop waiting
-                // and raise the exception, but the actual task is left running.
-
+                // .NET 6.0 added cancellation support to GetHostAddressesAsync, but it is only supported on Windows.
+                // For now, we keep the same cancellation handling for all TFM and OS platforms.
+                // However, the CA2016 quality rule forces us to specify a CancellationToken parameter.
 #if NET6_0_OR_GREATER
-                CancellationTokenSource? combinedCts = null;
-                try
-                {
-                    var finalCt = cancellationToken;
-
-                    if (timeout.IsSet)
-                    {
-                        combinedCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-                        combinedCts.CancelAfter((int)timeout.CheckAndGetTimeLeft().TotalMilliseconds);
-                        finalCt = combinedCts.Token;
-                    }
-
-                    return Dns.GetHostAddressesAsync(Host, cancellationToken);
-                }
-                finally
-                {
-                    combinedCts?.Dispose();
-                }
+                return Dns.GetHostAddressesAsync(Host, CancellationToken.None)
```

Comment:
Wait, we still want to pass the user cancellation token to GetHostAddressesAsync, no? (e.g. for Windows)? Otherwise, what's the point of calling this overload and having an #if in the first place?

Reply:
Note that in order for the timeout to work, Dns.GetHostAddressesAsync needs to receive a linked cancellation token that already has the timeout configured on it.

Reply:
Cancellation support added for Windows only.